### PR TITLE
docs: Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by faddat**
Updates the changelog in preparation for a new release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
 
+### Improvements
+
+- [#538](https://github.com/cosmos/iavl/pull/538) Use cosmos-db instead of tm-db
+- [#537](https://github.com/cosmos/iavl/pull/537) Add a gitpod development environment so it is easy to work with all supported databases
+
+
+**Linting and cleanup**
+
+- [#542](https://github.com/cosmos/iavl/pull/542) Move keyformat and fastnode to specific folders
```
Comment (robert-zaremba) [inline CHANGELOG.md:13]: this is a breaking change. Should be listed in `### Breaking Changes section`

**Review by robert-zaremba — COMMENTED** [review]
Please add Breaking Changes section.

File: CHANGELOG.md (hunk 1 above)
Comment (faddat) [reply CHANGELOG.md:13]: Makes a billion types of sense, thank you

Comment (faddat) [reply CHANGELOG.md:13]: keyformat change?

Or do you mean badger and bolt?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+- removal of remotedb
+
+
+**Linting and cleanup**
+
+- [#542](https://github.com/cosmos/iavl/pull/542) Move keyformat and fastnode to specific folders
+- [#535](https://github.com/cosmos/iavl/pull/535) Lint the iavl repository in ci
+- [#536](https://github.com/cosmos/iavl/pull/536) Make iavl satisfy golangci-lint conditions
+- [#540](https://github.com/cosmos/iavl/pull/540) Configure golangci-lint
+- [#543](https://github.com/cosmos/iavl/pull/543) Disable a few linters, and further cleanup
```
Comment (marbar3778) [inline CHANGELOG.md:23]:
```suggestion
```

this doesn't effect upstream users

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,26 @@
...
+### Improvements
+
+- [#538](https://github.com/cosmos/iavl/pull/538) Use cosmos-db instead of tm-db
+- [#537](https://github.com/cosmos/iavl/pull/537) Add a gitpod development environment so it is easy to work with all supported databases
+
+
+**breaking changes compared to tm-db**
+- removal of boltdb
+- removal of badgerdb
+- removal of remotedb
```
Comment (marbar3778) [inline CHANGELOG.md:14]:
```suggestion
```

File: CHANGELOG.md (hunk 1 above)
Comment (robert-zaremba) [reply CHANGELOG.md:19]: external API changed - you moved some functions / types to other package.

Comment (robert-zaremba) [reply CHANGELOG.md:12]: UP, this should go into Breaking Changes section

**Pull request closed without merging**